### PR TITLE
Space wizard battle

### PR DESCRIPTION
**Pull request by sensible-heart**
Implementation of space wizard attacks and audio that occurs during the battle. Just place space wizard object in the start room to test. Noted Bugs: Space wizard hammer animation plays too quick to see sometimes (polish issue that can be ignored if needed), space wizard can only be attacked properly with a weapon (can be feature).

### CODE REVIEW

**Comment by adamjford** [comment]
Want me to take a swing at the merge that's needed here?

**Comment by sensible-heart** [comment]
that would be great!

**Comment by adamjford** [comment]
After doing the merge and testing it, I found the following two things
- If you get directly between the boss and a wall (I specifically had it with the boss on your left and the wall on your right), you get stuck and can't move anyway
- I get the following error immediately after dying:

```
FATAL ERROR in
action number 1
of  Step Event0
for object obj_boss:

Unable to find any instance for object index '65' name 'obj_player'
 at gml_Object_obj_boss_StepNormalEvent_1 (line 1) - action_linear_step( obj_player.x, obj_player.y, 1, 0 );
```

Not 100% sure if these issues are due to my merge or your changes.

**Comment by sensible-heart** [comment]
You can't attack when you get stuck by a wall?
 I'm pretty sure that's just a missing if check in my code that I fixed in other enemy iterations, caused by the death animation code which wasn't in the version of master I branched off of.

**Comment by adamjford** [comment]
Well, I mean, the boss wasn't moving either, so I think they got stuck together.

Sounds like my merge didn't directly cause either issue, so I'll push it up.

**Comment by sensible-heart** [comment]
Great Adam, thanks! I'll take another look once I finish this robot implementation.

**Comment by sensible-heart** [comment]
Okay I believe I fixed  the fatal error. Should be good to test and then merge into master.

**Comment by adamjford** [comment]
Fixed a bug I found in my testing where, if a fireball spawned right after the player died, it caused an error trying to find the now non-existent player.

I also noticed that the boss can push you into a wall. Looking into that now.

File: dungeon.gmx/objects/obj_boss.object.gmx [hunk 1]
```diff
@@ -134,7 +357,51 @@ alarm[0] = 15;
...
+        <codestring></codestring>
+        <whoName>self</whoName>
+        <relative>0</relative>
+        <isnot>0</isnot>
+        <arguments>
+          <argument>
+            <kind>1</kind>
+            <string>with (other)
+{
+    direction = point_direction(other.x, other.y, x, y); //note that since with() pushes the code over to other, it is now the enemy that is "other" within that construct
```
Comment (adamjford) [inline dungeon.gmx/objects/obj_boss.object.gmx:440]: This is the code that causes the player to get pushed away, right? Not sure how this is causing the player to skip right into the wall, as it's just like any other movement, right? Thus, you'd think the player would just collide with the wall as usual.

**Pull request merged**